src/main/java/algorithms/curated170/easy/ShortestWordDistance.java

```diff
@@ -3,8 +3,7 @@
 public class ShortestWordDistance {
```

Comment:
We should change like this. But, it is not an efficient way, as far as I understand.
Because the "shortest" method call will be done too many times. 
![image](https://user-images.githubusercontent.com/60903744/118301104-5fa69900-b4eb-11eb-99a3-4cb5a1c01d00.png)

The other approach uses precomputation. It is why it was more appropriate for this case.
But maybe we can find a way to handle with this approach too.

 
```
class WordDistance {
    String[] words;
    public WordDistance(String[] wordsDict) {
         this.words = wordsDict;
    }
    
    public int shortest(String word1, String word2) {
      
        int minDist = words.length, i1 = minDist, i2 = -minDist;
        for (int i = 0; i < words.length; i++) {
        if (words[i].equals(word1)) {
            i1 = i;
        } else if (words[i].equals(word2)) {
            i2 = i;
        }

       
            minDist = Math.min(minDist, Math.abs(i1 - i2));
        
    }
    return minDist;
    }
     
}

```

Reply:
Wasn't the original question just about creating a method and not a class?

Reply:
I updated the issue with images and the default code.
Sorry for the late consideration.
https://github.com/spiralgo/algorithms/issues/42

Reply:
I don't think looping through all the indices of the words array can be or be improved to be efficient like the map solution.

Reply:
Yeah, you are right.
Maybe for the nonce, we can retreat and remove this approach until the time we find an efficient way.
As you know, sometimes we should lose the battles to win the war.

Reply:
What should we upload now then?

Reply:
The other approach is working quite fine.

Reply:
Are we done with that approach or should we change anything further?

Reply:
I mean we can remove ShortestWordDistance.java until we can find an efficient way to do this without precomputation (at least enough to pass the test cases.)

The precomputation approach works nice.

Reply:
> I mean we can remove ShortestWordDistance.java until we can find an efficient way to do this without precomputation (at least enough to pass the test cases.)
> 
> The precomputation approach works nice.
> 

The other file is the easy version of the question without classes.

Reply:
But it works?
![image](https://user-images.githubusercontent.com/60903744/118363731-7b1eac00-b59e-11eb-93b1-0e27a062b088.png)

Reply:
The question is this one, right?
https://leetcode.com/problems/shortest-word-distance-ii/submissions/

Reply:
Yes.
However, I have also added a small change to [this question](https://github.com/spiralgo/algorithms/issues/147). That is the file that you have started this conversation in.

Reply:
I shall merge then if you have also understood the situation.

Reply:
Oke, we can merge. Thank you, Erdem.